pkg/experiment/metadata.go

```diff
@@ -124,15 +141,40 @@ func sslOptions(config MetadataConfig) *gocql.SslOptions {
 }
 
 // connect creates a session to the Cassandra cluster. This function should only be called once.
-func (m *Metadata) connect() error {
+func (m *Metadata) connect() *gocql.ClusterConfig {
 	cluster := gocql.NewCluster(m.config.CassandraAddress)
 
 	// TODO(niklas): make consistency configurable.
 	cluster.Consistency = gocql.LocalOne
 	cluster.SerialConsistency = gocql.LocalSerial
 
 	cluster.ProtoVersion = 4
-	cluster.Timeout = m.config.CassandraConnectionTimeout
+	cluster.ConnectTimeout = m.config.CassandraConnectionTimeout
+	cluster.Timeout = m.config.CassandraTimeout
+	cluster.IgnorePeerAddr = m.config.CassandraIgnorePeerAddr
+	cluster.DisableInitialHostLookup = !m.config.CassandraInitialHostLookup
+
+	return cluster
+}
+
+func (m *Metadata) createKeyspace() error {
+	config := m.connect()
+	session, err := config.CreateSession()
+	defer session.Close()
+	if err != nil {
+		return fmt.Errorf("cannot achieve session for creating keyspace: %q", err.Error())
+	}
+
+	query := fmt.Sprintf("CREATE KEYSPACE IF NOT EXISTS %s WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor': 1};", m.config.CassandraKeyspaceName)
+
+	return session.Query(query).Exec()
+
+}
+
+// Connect creates a session to the Cassandra cluster. This function should only be called once.
+func (m *Metadata) Connect() error {
```

Comment:
Do you think it would be wise to `connect` during Metadata object creation? So that initialized object would be ready to use.